test/integration/local_lambda/test_end_to_end.py

```diff
@@ -84,148 +70,170 @@ def invoke_function(self, port):
             f"http://localhost:{port}/2015-03-31/functions/function/invocations", json={}
         )
         
-    def create_container_and_invoke_function(self, cmd, port):
-        self.run_command(cmd)
-        
-        # sleep 1s to give enough time for the endpoint to be up to curl
-        self.sleep_1s()
-        
-        return self.invoke_function(port)
-
-    @parameterized.expand([("x86_64", "8000"), ("arm64", "9000"), ("", "9050")])
-    def test_env_var_with_equal_sign(self, arch, port):
+    @contextmanager
+    def create_container(self, cmd, image):
+        try:
+            platform = "x86_64" if self.ARCH == "" else self.ARCH
+            cmd_full = f"docker run --platform linux/{platform} {cmd}"
+            self.run_command(cmd_full)
+        
+            # sleep 1s to give enough time for the endpoint to be up to curl
+            self.sleep_1s()
+            yield
+        except Exception as e:
+            print(f"An error occurred while executing cmd: {cmd_full}. error: {e}")
+            raise e
+        finally:
+            self.run_command(f"docker stop {image}")
+            self.run_command(f"docker rm -f {image}")
+        
+
+    def test_env_var_with_equal_sign(self, arch=ARCH, port=PORT):
```

Comment:
if you're never passing the variable, I think it's clearer to just define these as variables inside the function
```
def test_env_var_with_equal_sign(self):
  arch = ARCH
  port = PORT
```

(maybe not even define it as a local variable and just use it as `ARCH` or `PORT` whenever you need it)

Reply:
sure we can do it that way